engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/batch/NadelBatchHydrationByIndex.kt

```diff
@@ -75,21 +117,22 @@ internal object NadelBatchHydrationByIndex {
         private val values: List<Any?> = getValues(batches)
         private var cursorIndex = 0
 
-        fun take(): Any? {
+        fun take(inputValues: List<Any?>): List<Any?> {
+            return inputValues.map(::takeOne)
+        }
+
+        fun takeOne(inputValue: Any?): Any? {
+            if (inputValue == null) {
```

Comment:
Return null if the input node was null (i.e. null means not sent to underlying service). Also does NOT advance the cursor for the list hydration results.